feat(config)!: migrate zebrad to use a layered configuration (#9768)
ZcashFoundation/zebra: book/src/user/docker.md

## Configuring Zebra

> Zebra uses config-rs to layer configuration from defaults, an optional TOML file, and `ZEBRA_`-prefixed environment variables. When running with Docker, configure Zebra using any of the following (later items override earlier ones):

Reviewer: ```suggestion
Zebra uses [config-rs](https://crates.io/crates/config) to layer configuration from defaults, an optional TOML file, and `ZEBRA_`-prefixed environment variables. When running with Docker, configure Zebra using any of the following (later items override earlier ones):
```

1. **Provide a specific config file path:** Set the `CONFIG_FILE_PATH` environment variable to point to your config file within the container. The entrypoint will pass it to `zebrad` via `--config`.
2. **Use the default config file:** Mount a config file to `/home/zebra/.config/zebrad.toml` (for example using the `configs:` mapping in `docker-compose.yml`). This file is loaded if `CONFIG_FILE_PATH` is not set.
3. **Use environment variables:** Set `ZEBRA_`-prefixed environment variables to override settings from the config file. Examples: `ZEBRA_NETWORK__NETWORK`, `ZEBRA_RPC__LISTEN_ADDR`, `ZEBRA_RPC__ENABLE_COOKIE_AUTH`, `ZEBRA_RPC__COOKIE_DIR`, `ZEBRA_METRICS__ENDPOINT_ADDR`, `ZEBRA_MINING__MINER_ADDRESS`.

You can verify your configuration by inspecting Zebra's logs at startup.

### RPC

> Zebra's RPC server is disabled by default. Enable and configure it via TOML or environment variables:

Reviewer: Nitpick/Optional:

```suggestion
Zebra's RPC server is disabled by default. Enable and configure it via the TOML configuration file, or configuration environment variables:
```

```

Note that Docker will run Zebra with the RPC server enabled and the cookie

> authentication mechanism disabled since Lightwalletd doesn't support it. In this

Reviewer: Nitpick: It would be nice to be more specific here.

```suggestion
authentication mechanism disabled when running `docker compose -f docker/docker-compose.lwd.yml up`, since Lightwalletd doesn't support cookie authentication. In this
```
